bin/propolis-server/src/main.rs

```diff
@@ -21,6 +23,49 @@ use propolis_server::{
 /// Threads to spawn for tokio runtime handling the API (dropshot, etc)
 const API_RT_THREADS: usize = 4;
 
+/// Configuration for metric registration.
+#[derive(Clone, Debug)]
+enum MetricRegistration {
+    Disable,
+    Dns,
+    WithAddr(SocketAddr),
+}
+
+impl fmt::Display for MetricRegistration {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            MetricRegistration::Disable => "disable".fmt(f),
+            MetricRegistration::Dns => "dns".fmt(f),
+            MetricRegistration::WithAddr(addr) => addr.fmt(f),
+        }
+    }
+}
+
+impl FromStr for MetricRegistration {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "disable" => Ok(Self::Disable),
+            "dns" => Ok(Self::Dns),
```

Comment:
nitpicky, take it or leave it: should we, perhaps, use [`str::eq_ignore_ascii_case`](https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case) here, in the interest of "be conservative in what you emit, liberal in what you accept"?

Reply:
Done in 56792ba